### PR TITLE
Refactor setup commands and add unit tests

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
Refactor setup commands and add unit tests

### CODE REVIEW

File: components/cli/cli/cli.go [hunk 1]
```diff
@@ -50,6 +51,7 @@ type Cli interface {
 	CredManager() credentials.CredManager
 	CredReader() credentials.CredReader
 	Runtime() cliRuntime.Runtime
+	Sleep(seconds time.Duration)
```
Comment (Mirage20) [inline components/cli/cli/cli.go]: Why do we need Sleep as an interface function? Can't we directly use time.Sleep?

File: components/cli/cmd/cellery/setup_cleanup_existing.go [hunk 2]
```diff
@@ -35,7 +36,7 @@ func newSetupCleanupExistingCommand() *cobra.Command {
 		Short: "Cleanup existing cluster setup",
 		Args:  cobra.NoArgs,
 		Run: func(cmd *cobra.Command, args []string) {
-			setup.RunCleanupExisting(knative, istio, ingress, hpa, confirmed)
+			setup.RunCleanupExisting(cli, knative, istio, ingress, hpa, confirmed)
```
Comment (Mirage20) [inline components/cli/cmd/cellery/setup_cleanup_existing.go]: Does this returns an error?

File: components/cli/pkg/commands/setup/modify.go [hunk 3]
```diff
@@ -111,65 +204,83 @@ func modifyRuntime(cli cli.Cli) {
...
 		{
-			applyChanges(cli)
-			return
+			return applyChanges(cli)
 		}
 	default:
 		{
 			RunSetup(cli)
-			return
+			return nil
```
Comment (Mirage20) [inline components/cli/pkg/commands/setup/modify.go]: Does RunSetup returns an error?

File: components/cli/cli/cli.go (hunk 1 above)
Comment (MadhukaHarith92) [reply components/cli/cli/cli.go]: Previously we directly used time.Sleep. However now when this function is getting called from unit tests, they are being kept waiting for that same period of time unnecessarily.

File: components/cli/cmd/cellery/setup_cleanup_existing.go (hunk 2 above)
Comment (MadhukaHarith92) [reply components/cli/cmd/cellery/setup_cleanup_existing.go]: Updated.

File: components/cli/pkg/commands/setup/modify.go (hunk 3 above)
Comment (MadhukaHarith92) [reply components/cli/pkg/commands/setup/modify.go]: Updated in all places.

**Pull request merged**